### DISCUSSION TITLE
Where in the documentation is built in metrics? I'm curious how Volume Weighted Average Price is implemented. I can see SMA mentioned in the docs, but not sure what else is availalbe.

### DISCUSSION THREAD

**Discussion by branhoff** [kernc/backtesting.py#885] [Q&A]
I tried to do my own analysis of VWAP vs SMAP years ago. I recently picked it back up, but the standard for loop is CRAZY slow. I've been playing around with numpy methods to by pass traditional python looping.

I can't find where these standard metrics are. Is the point of this library to build from scratch?

**Comment by kernc** [comment]
From [Quick Start User Guide](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Strategy):
> Backtesting.py doesn't ship its own set of technical analysis indicators. Users favoring TA should probably refer to functions from proven indicator libraries, such as [TA-Lib](https://github.com/mrjbq7/ta-lib) or [Tulipy](https://tulipindicators.org/) ...

So no VWAP built in.

**Reply by BradKML** [reply]
Wait VWAP... are there anyway to get this in? But I guess Panads-TA has them https://github.com/twopirllc/pandas-ta
